generator/lookml.py

```diff
@@ -204,16 +283,39 @@ def _lookml(namespaces, glean_apps, target_dir, dryrun, namespace_filter=[]):
     help="Use the Cloud Function to run dry runs during LookML generation.",
     type=bool,
 )
+@click.option(
+    "--parallelism",
+    "-p",
+    default=8,
+    type=int,
+    help="Number of threads to use for lookml generation",
+)
 def lookml(
-    namespaces, app_listings_uri, target_dir, metric_hub_repos, only, use_cloud_function
+    namespaces,
+    app_listings_uri,
+    target_dir,
+    metric_hub_repos,
+    only,
+    use_cloud_function,
+    parallelism,
 ):
     """Generate lookml from namespaces."""
     if metric_hub_repos:
         MetricsConfigLoader.update_repos(metric_hub_repos)
-
     glean_apps = _get_glean_apps(app_listings_uri)
 
-    dryrun = functools.partial(
-        DryRun, bigquery.Client(), use_cloud_function, id_token()
+    dry_run_id_token = None
+    if use_cloud_function:
+        dry_run_id_token = id_token()
+
+    dryrun = functools.partial(DryRun, use_cloud_function, dry_run_id_token)
```

Comment:
Passing this partial `DryRun` constructor function across module boundaries still seems less than ideal (e.g. there's no great way to type hint it).  Maybe this could be better structured using some new class that's initialized with just the parts we want shared and has a method to do/create the dryrun (e.g. `DryRunContext` or `DryRunFactory`)?

Reply:
mh, it is possbile. But then calls would look something like `dryrun.create(sql=..., table=...).get_schema()`

Reply:
I added a separate commit (the last one) with a `DryRunContext`. I think it's probably a bit more readable